inc/search.class.php

```diff
@@ -567,11 +645,14 @@ static function prepareDatasForSearch($itemtype, array $params, array $forcedisp
     * @since 0.85
     *
     * @param array $data  Array of search datas prepared to generate SQL
+    * @param CommonDBTM|false $sub_item Are we calling from a sub item?
     *
     * @return void
+    *
+    * @since 10.0.0 Method is no longer static
    **/
-   static function constructSQL(array &$data) {
-      global $DB, $CFG_GLPI;
+   public function constructSQL(array &$data, $sub_item = false) {
```

Comment:
1. For sublists, `$sub_item` is an array. Maybe its declaration should be `?array $sub_item = null` to be able to introduce a strong type.

2. This method is used in some plugins (`formcreator` and `fusioninventory` for instance), its renaming and its parameter changes should be documented.

3. There is still 1 occurence of static usage of `Search::constructSQL()` in GLPI code.